assets/wizards/readerRevenue/views/stripe-setup/webhooks-settings.js

```diff
@@ -0,0 +1,193 @@
+									} )
+								}
+							/>
+							<code className="mr3">{ webhook.url }</code>
+						</span>
+						<Button
+							isDestructive
+							isLink
+							disabled={ isLoading }
+							onClick={ () => {
+								if (
+									utils.confirmAction(
+										__( 'Are you sure you want to remove this webhook?', 'newspack' )
+									)
+								) {
+									handleChange( webhook, 'DELETE' );
+								}
+							} }
+						>
+							{ __( 'Delete', 'newspack' ) }
+						</Button>
+					</li>
+				) ) }
+			</ul>
+		</div>
+	);
+};
+
+const WebhooksSettings = () => {
+	const [ webhooksList, setWebhooksList ] = useState( [] );
+	const { wizardApiFetch } = useDispatch( Wizard.STORE_NAMESPACE );
+	useEffect( () => {
+		wizardApiFetch( {
+			path: `/newspack/v1/stripe/webhooks`,
+		} ).then( setWebhooksList );
+	}, [] );
+
+	const [ thisSiteWebhooks, otherWebhooks ] = webhooksList.reduce(
+		( acc, webhook ) => {
+			if ( webhook.matches_url ) {
+				acc[ 0 ].push( webhook );
+			} else {
+				acc[ 1 ].push( webhook );
+			}
+			return acc;
+		},
+		[ [], [] ]
+	);
+	const activeThisSiteWebhooks = thisSiteWebhooks.filter( webhook => webhook.status === 'enabled' );
+
+	const handleWebhooksListUpdate = method => payload => {
+		let newList = webhooksList;
+		switch ( method ) {
+			case 'POST':
+				newList = webhooksList.map( webhook => ( payload.id === webhook.id ? payload : webhook ) );
+				break;
+			case 'DELETE':
+				newList = webhooksList.filter( webhook => payload.id !== webhook.id );
+				break;
+		}
+		setWebhooksList( newList );
+	};
+	const resetWebhooks = () => {
+		const args = {
+			path: `/newspack/v1/stripe/reset-webhook`,
+			method: 'DELETE',
+			isQuietFetch: true,
+		};
+		wizardApiFetch( args ).then( setWebhooksList );
+	};
+	return (
+		<div className="newspack-payment-setup-screen__webhooks">
+			{ activeThisSiteWebhooks.length > 1 && (
+				<Notice
+					isError
+					noticeText={ __(
+						'There are too many webhoooks for this site. Please delete or disable the extra ones.',
+						'newspack'
+					) }
+				/>
+			) }
+			{ activeThisSiteWebhooks.length === 0 && (
+				<Notice
+					isError
+					noticeText={ __( 'There are no active webhoooks for this site.', 'newspack' ) }
+				/>
+			) }
+			<ActionCard
+				title={ __( 'Webhooks', 'newspack' ) }
+				className="newspack-settings__no-border"
+				description={ __(
+					'Manage the webhooks Stripe uses to communicate with your site.',
+					'newspack'
+				) }
+			/>
+			<WebhooksList
+				withBorder
+				className="mb4"
+				title={ () => (
+					<div className="flex justify-between items-center">
+						<h4 className="b f6 ma0">{ __( 'Webhooks connected to this site', 'newspack' ) }</h4>
+						<Button
+							variant="secondary"
+							isSmall
+							onClick={ () => {
+								if (
+									utils.confirmAction(
+										__( 'Are you sure you want to reset all webhooks?', 'newspack' )
```

Comment:
```suggestion
										__( 'Are you sure you want to reset all webhooks connected to this site?', 'newspack' )
```

This only deletes and recreates the webhooks connected to the site, right?